src/System.Runtime.Extensions/src/System/IO/PathHelper.Windows.cs

```diff
@@ -223,15 +191,15 @@ private StringBuilder GetFullPathName(string path)
                 Debug.Assert(!PathInternal.IsExtended(path));
             }
 
-            StringBuilder outputBuffer = this.GetOutputBuffer(capacity);
+            StringBuffer outputBuffer = new StringBuffer((ulong)capacity);
```

Comment:
This is an example of my asking about why we're pooling the underlying thing rather than the top-level thing.  If we're going to pool, why not pool StringBuffer instead so we can avoid this allocation?